lekce/3/javascript.md

```diff
@@ -1,73 +1,299 @@
+
+Otevři si prohlížeč Google Chrome a použij klávesovou zkratku <kbd>Ctrl + Shift + J</kbd> (na Macu <kbd>Cmd + Alt + J</kbd>). Měla by se ti otevřít _vývojářská konzole_, kde můžeme spouštět JavaScript.
+
+<figure class="image"><img src="console.png" style="max-height:70vh;width:auto;" alt="konzole v Google Chrome"></figure>
+
+Tam, kde ti bliká kurzor, vlož následující kód:
+
+```js
+console.log('Hello, ' + window.prompt('Please, enter your name') + '!');
+```
+
+Stiskni <kbd>Enter</kbd>. Stránka se tě zeptá na jméno (funkce `window.prompt()`) a vypíše jej s pozdravem (funkce `console.log`).
+
+Můžeš zkusit vložit i jednořádkový kód, jenž jsme si ukázali v předchozí sekci tohoto návodu, a ověřit tak, že opravdu i bez odsazování bude program fungovat. Dostaneš výsledek volání funkce `console.log(isUkulele(6))`, tedy `false` (ukulele má struny čtyři, ne šest).
+
+----
+
+<!-- .slide: data-state="c-slide-task" -->
+
+## Cvičení
+
+Zkoušej zadávat i další věci. Do konzole lze vypisovat pomocí `console.log()`, ale budou fungovat i výrazy jako `7 * 6` a i další věci, které už znáš z Pythonu. Nemusíš psát vše na jeden řádek, konzole podporuje i víceřádkový vstup a dokonce se ti pokusí i napovídat (nápovědy lze odsouhlasit a použít pomocí <kbd>Tab</kbd>), podobně jako to umí `ipython`.
+
+Všimni si, že nad konzolí je menu, přes které se můžeš dostat i na další vývojářské nástroje. Ty slouží k tomu, abys mohla různě prozkoumávat aktuálně načtenou webovou stránku. Konzole a i všechny další vývojářské nástroje v Google Chrome [mají svou dokumentaci][chrome-console], kde se můžeš dovědět více.
+
+[chrome-console]: https://developers.google.com/web/tools/chrome-devtools/
 
 ---
 
-## Knihovna jQuery
+### Proměnné
 
 ----
 
-Možná v ní jen napsat skoro to samý 
+Když si chceme v Pythonu nastavit nějakou novou proměnnou, stačí nám udělat toto:
+
+```python
+strings_count = 4
+```
+
+Proměnná bez hodnoty nemůže existovat. V JavaScriptu je to trochu jinak. Máme klíčové slovo `var` (jako _variable_, angl. proměnná), jímž v programu říkáme: „Vytvoř mi novou proměnnou s tímto názvem.“ Volitelně můžeme přidat i počáteční hodnotu. Ukažme si to třeba na funkci, která z hodiny umí poznat, zda je den nebo noc.
+
+```js
+function getTimePeriod(now) {
+    // nová proměnná, bez hodnoty
+    var timePeriod;
+
+    // nová proměnná, s počáteční hodnotou
+    var hour = now.getHours();
+
+    if (hour > 6 && hour < 22) {
+        timePeriod = 'day';
+    } else {
+        timePeriod = 'night';
+    }
+
+    return timePeriod;
+}
+
+var now = new Date();
+console.log(getTimePeriod(now));
+```
+
+Proměnnou `timePeriod` jsme vytvořili prázdnou, a až později jsme do ní přiřadili nějakou hodnotu. Proměnnou `hour` jsme ale vytvořili už rovnou s nějakou počáteční hodnotou. Způsob, jakým v JavaScriptu fungují proměnné, je složitější, ale toto nám bude prozatím stačit.
 
```

Comment:
Uvažuju, jestli by nestálo za to se zmínit o lexikálním scopu a hoistingu, jelikož to může začátečníkům způsobit hodně frustrace a bolení hlavy.